CRM/Campaign/Form/Gotv.php

```diff
@@ -54,7 +54,7 @@ public function preProcess() {
     }
     $this->assign('force', $this->_force);
     $this->assign('votingTab', $this->_votingTab);
-    $this->assign('searchParams', json_encode($this->get('searchParams')));
+    $this->assignScriptVar('searchParams', $this->get('searchParams'));
```

Comment:
I'm a little surprised how common this pattern is -- e.g. subjectively, to me, it feels more natural to assign the original data (e.g. `$searchParams`) and let then the view-template decide what kind of encodings to apply.

What I'm not sure about is... all these controllers correspond to templates, and the templates can use variables in a few ways -- e.g. Outputting `<script>`-JSON might be different from outputting attribute-JSON. Anecdotally, it feels like `<script>`-JSON is the common thing... but I can't easily search/confirm.

`assignScriptVar()` is using `JSON_HEX_TAG | JSON_UNESCAPED_SLASHES`, which his good for `<script>`. But if there are attribute-JSON instances, could they have trouble with `JSON_UNESCAPED_SLASHES`?

Reply:
Wouldn't `Civi::resources()->addVars('myNamespace', ['foo' => 'bar']);` be much preferable to assigning a smarty variable to then render in a `<script>`? Shouldn't we be discouraging `<script>` in Smarty templates altogether?